File: evidence/generated-systems/golang/insurance-on-golang/insurance-on-golang/internal/model/Exposure.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// Exposure Declaration
//==============================================================
type Exposure struct {
    gorm.Model
     ClaimId         *uint
    Claim           *Claim `gorm:"foreignKey:ClaimId"`
    PolicyCoverageId         *uint
    PolicyCoverage           *PolicyCoverage `gorm:"foreignKey:PolicyCoverageId"`
    InsuredObjectId         *uint
    InsuredObject           *InsuredObject `gorm:"foreignKey:InsuredObjectId"`
     Reserves           []ClaimReserve `gorm:"foreignKey:ReservesFromExposureId"`
     Payments           []ClaimPayment `gorm:"foreignKey:PaymentsFromExposureId"`
    ExposureType                      ExposureType
    Status                      ExposureStatus

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/insurance-on-rails/insuranceonrails/app/models/exposure.rb
class Exposure < ApplicationRecord
  enum ExposureType: [:BodilyInjury, :PropertyDamage, :Medical, :UninsuredMotorist, :PersonalInjuryProtection, :DwellingDamage, :ContentsDamage, :BusinessIncome]
  enum Status: [:Open, :Closed, :Pending, :Reserved]


  has_many :Claim, class_name: 'Claim'
  has_many :PolicyCoverage, class_name: 'PolicyCoverage'
  has_many :InsuredObject, class_name: 'InsuredObject'
  has_many :Reserves, class_name: 'ClaimReserve'
  has_many :Payments, class_name: 'ClaimPayment'

end
